articles/modules/ROOT/pages/how-to-force-direction-of-expansion.adoc

```diff
@@ -0,0 +1,128 @@
+
+https://neo4j.com/developer/kb/how-to-avoid-costly-traversals-with-join-hints/[How to avoid costly traversals with join hints]
+
+=== Label filtering without anchoring
+
+The presence of a label on a node in the Match pattern can significantly change whether or not the planner will try to use it as an anchor node.
+
+That is, the planner prefers index lookups to find anchor nodes, though it may still use label scans to find anchor nodes (especially when there are no opportunities for index lookups in the query).
+
+Neither index lookups or label scans are possible when the label isn't present in the pattern (or otherwise described in a Where clause, like `WHERE c:Country`), and these two means of lookup are the two most common for finding anchor nodes in the graph.
+Of course, you may need to have that label in the query for the sake of correctness, otherwise the wrong nodes and patterns might be matched.
+
+There is a way to perform the label filtering in a special manner which also prevents the planner from using it for index lookup or label scanning.
+Here is an example:
+
+[source,cypher]
+----
+MATCH (user:User {id:12345})-[:LIVES_IN]->(country {name:'United States'})
+WHERE 'Country' IN labels(country)
+----
+
+Note that we removed the `:Country` label from the Match pattern, preventing the planner from being able to use it as an anchor node via an index lookup or label scan.
+The planner has no choice but to use the `user` node as an anchor, and expand relationships to the `country` node at the other end.
+
+The second line, this membership check of a certain label being present in the node's labels, lets us still retain the label filtering we need for correctness.
+
+Note that attempting to use `WHERE country:Country` to accomplish the same thing will not work, as the planner is aware of this syntax and can still use labels  used here for lookups to find anchor nodes.
+
+=== Node identity filtering to avoid anchoring
+
+If the end nodes are known such that they can be efficiently pre-matched (such as via index lookup), when we want to prevent one from being used as an anchor for expansion, then we can use this trick.
+
+Consider this query:
+
+[source,cypher]
+----
+MATCH (user:User {id:12345}), (country:Country {name:'United States'})
+MATCH (user)-[:LIVES_IN]->(c)
+WHERE c = country
+----
+
+The key is in the second and third lines, usage of the unlabeled variable `c`.
+
+Even though we have matched to both end nodes, and they are both potential anchors, in the second Match  it is clear that the `user` node is the only one we can expand from; the planner is only aware that `c` is an unlabeled node and not a candidate for an anchor node.
+
+The filtering that the `c` node we expand to must be the same as the `country` anchor node is something the planner can only consider after the expansion is finished, so there is no oppportunity for it to use `country` as an anchor for expansion.
```

Comment:
the planner usually does a hash join here checking the expanded end node against the set of "country" nodes.